notes.md

```diff
@@ -0,0 +1,264 @@
+# Design Notes
+
```

Comment:
Caveat: I developed and ran this on another repo where the folder structure was slightly different (and the apps were named differently), so I can't guarantee the examples below will run out of the box on this repo. We will at least need migration files to be generated to get the examples to run in this PR.